[CI] ShardRoutingTests testEqualsIgnoringVersion failing

**Build Scans:**
- [elasticsearch-intake #44420 / part1](https://gradle-enterprise.elastic.co/s/axbubi67ci5tc)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.cluster.routing.ShardRoutingTests.testEqualsIgnoringVersion" -Dtests.seed=802A2B30D24E78A4 -Dtests.locale=nl-AW -Dtests.timezone=Etc/GMT+10 -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.ShardRoutingTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEqualsIgnoringVersion'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected non-equality
this  [test][0], node[null], [R], recovery_source[peer recovery], recovery_priority[UNASSIGNED_NEW], s[UNASSIGNED], unassigned_info[[reason=INDEX_CLOSED], at[+77619617-06-03T23:31:17.538Z], delayed=false, details[auto generated for test], allocation_status[no_attempt]], failed_attempts[0],
other [test][0], node[null], [R], recovery_source[peer recovery], recovery_priority[UNASSIGNED_NEW], s[UNASSIGNED], unassigned_info[[reason=INDEX_CREATED], at[2026-07-23T18:28:12.480Z], delayed=false, details[auto generated for test_1], allocation_status[no_attempt]], failed_attempts[0]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testEqualsIgnoringVersion
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -588,9 +588,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:change_point.values null column}
   issue: https://github.com/elastic/elasticsearch/issues/154877
-- class: org.elasticsearch.cluster.routing.ShardRoutingTests
-  method: testEqualsIgnoringVersion
-  issue: https://github.com/elastic/elasticsearch/issues/154886
 - class: org.elasticsearch.xpack.core.ml.datafeed.DatafeedUpdateTests
   method: testIsUserInitiatedProjectRoutingChangeWhenRoutingNarrowsShouldReturnTrue
   issue: https://github.com/elastic/elasticsearch/issues/154908

--- a/server/src/test/java/org/elasticsearch/cluster/routing/ShardRoutingTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/routing/ShardRoutingTests.java
@@ -454,6 +454,7 @@ public class ShardRoutingTests extends AbstractWireSerializingTestCase<ShardRout
                     otherRouting.primary(),
                     otherRouting.state()
                 ).withRelocatingNodeId(otherRouting.relocatingNodeId())
+                    .withRecoveryPriority(otherRouting.recoveryPriority())
                     .withUnassignedInfo(
                         otherRouting.unassignedInfo() == null
                             ? new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, "test")

--- a/server/src/test/java/org/elasticsearch/cluster/routing/ShardRoutingTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/routing/ShardRoutingTests.java
@@ -392,6 +392,8 @@ public class ShardRoutingTests extends AbstractWireSerializingTestCase<ShardRout
                     if (otherRouting.started()) {
                         unchanged = true;
                     } else {
+                        final var originalState = otherRouting.state();
+                        final var originalRelocatingNodeId = otherRouting.relocatingNodeId();
                         otherRouting = new ShardRouting(
                             otherRouting.shardId(),
                             otherRouting.currentNodeId(),
@@ -399,7 +401,10 @@ public class ShardRoutingTests extends AbstractWireSerializingTestCase<ShardRout
                             otherRouting.primary(),
                             otherRouting.state(),
                             otherRouting.recoverySource(),
-                            TestShardRouting.buildRecoveryPriority(otherRouting.state(), otherRouting.relocatingNodeId() != null),
+                            randomValueOtherThan(
+                                otherRouting.recoveryPriority(),
+                                () -> TestShardRouting.buildRecoveryPriority(originalState, originalRelocatingNodeId != null)
+                            ),
                             otherRouting.unassignedInfo(),
                             otherRouting.relocationFailureInfo(),
                             otherRouting.allocationId(),

